### PR TITLE
made a simple view with search form for HPO terms

### PR DESCRIPTION
**Pull request by northwestwitch**
image

### CODE REVIEW

**Comment by northwestwitch** [comment]
The HPO term search on this page does not return auto-completing HPO terms. I did it on purpose because I think it might be useful for the user to have a free search and everything matching that search us returned. Otherwise the auto-completing one is always at the case level. What do you guys think? @moonso @dnil

**Comment by moonso** [comment]
Hi @northwestwitch , I've made some small changes including a better text search. Please try 🦑

**Comment by northwestwitch** [comment]
I'm testing your code but when I try to look for a term it crashes with this message:

2018-06-20 07:47:54 ChiaraRMBP.scilifelab.se flask.app[97891] ERROR Exception on /phenotypes [POST]
Traceback (most recent call last):
  File "/Users/chiararasi/miniconda3/envs/py36/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/chiararasi/miniconda3/envs/py36/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/chiararasi/miniconda3/envs/py36/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/chiararasi/miniconda3/envs/py36/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/Users/chiararasi/miniconda3/envs/py36/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/chiararasi/miniconda3/envs/py36/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/chiararasi/Documents/work/GITs/scout_fixes/scout/scout/server/utils.py", line 19, in decorated_function
    context = f(*args, **kwargs)
  File "/Users/chiararasi/Documents/work/GITs/scout_fixes/scout/scout/server/blueprints/phenotypes/views.py", line 20, in hpo_terms
    data = controllers.hpo_terms(store= store,  query = search_term, limit=limit)
  File "/Users/chiararasi/Documents/work/GITs/scout_fixes/scout/scout/server/blueprints/phenotypes/controllers.py", line 19, in hpo_terms
    hpo_phenotypes['phenotypes'] = list(store.hpo_terms(text=query, limit=limit))
  File "/Users/chiararasi/Documents/work/GITs/scout_fixes/scout/scout/adapter/mongo/hpo.py", line 103, in hpo_terms
    LOG.info("Found {0} terms with search word {1}".format(res.count(), search_term))
  File "/Users/chiararasi/miniconda3/envs/py36/lib/python3.6/site-packages/pymongo/cursor.py", line 755, in count
    cmd, self.__collation, session=self.__session)
  File "/Users/chiararasi/miniconda3/envs/py36/lib/python3.6/site-packages/pymongo/collection.py", line 1511, in _count
    session=session)
  File "/Users/chiararasi/miniconda3/envs/py36/lib/python3.6/site-packages/pymongo/collection.py", line 245, in _command
    retryable_write=retryable_write)
  File "/Users/chiararasi/miniconda3/envs/py36/lib/python3.6/site-packages/pymongo/pool.py", line 517, in command
    collation=collation)
  File "/Users/chiararasi/miniconda3/envs/py36/lib/python3.6/site-packages/pymongo/network.py", line 125, in command
    parse_write_concern_error=parse_write_concern_error)
  File "/Users/chiararasi/miniconda3/envs/py36/lib/python3.6/site-packages/pymongo/helpers.py", line 145, in _check_command_response
    raise OperationFailure(msg % errmsg, code, response)
pymongo.errors.OperationFailure: text index required for $text query

File: scout/adapter/mongo/hpo.py [hunk 1]
```diff
@@ -72,22 +73,34 @@ def hpo_terms(self, query=None, hpo_term=None, limit=None):
...
                     {'description': {'$regex': query, '$options':'i'}},
-                ]
+                ]   
             }
+            search_term = query
+        elif text:
+            new_string = ''
+            for i,word in enumerate(text.split(' ')):
+                if i == 0:
+                    new_string += word
```
Comment (northwestwitch) [inline scout/adapter/mongo/hpo.py]: new_string = word 
would work as well

File: scout/adapter/mongo/hpo.py [hunk 2]
```diff
@@ -72,22 +73,34 @@ def hpo_terms(self, query=None, hpo_term=None, limit=None):
...
         elif hpo_term:
             query_dict['hpo_id'] = hpo_term
+            search_term = hpo_term
 
-        if limit:
-            res = self.hpo_term_collection.find(query_dict).limit(limit)
-        else:
-            res = self.hpo_term_collection.find(query_dict)
+        limit = limit or int(10e10)
+        res = self.hpo_term_collection.find(query_dict).limit(limit).sort('hpo_number',ASCENDING)
```
Comment (northwestwitch) [inline scout/adapter/mongo/hpo.py]: Tried to print what the query_dict looks like when I search for two terms, for instance "bladder abnormality" and "bladder abn". 
In the first case the search is the following: {'$text': {'$search': 'bladder "abnormality"'}}, and gives 1085 results
in the second case it is this one: {'$text': {'$search': 'bladder "abn"'}}, and returns just 3 results.

I believe that the multiple word search is a great improvement for the tool, but wouldn't it be even more useful if it returns everything matching both expressions? I mean the second search should return at least as many results as the first one. Would using regular expressions be much slower?

File: scout/server/blueprints/phenotypes/templates/phenotypes/hpo_terms.html [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      Limiting the search to {{ phenotypes|length }} HPO phenotypes
+    </div>
+  </div>
+
+  <div class="row">
+    <div class="col-md-12">
+      <div class="panel panel-default">
+        <div class="panel-heading">HPO terms</div>
+          {% if phenotypes|length == 0 %}
+            The search didn't return ant phenotype term
```
Comment (northwestwitch) [inline scout/server/blueprints/phenotypes/templates/phenotypes/hpo_terms.html]: I made a typo. if you are going to fix the code could you fix this as well?

**Pull request merged**